src/main/scala/net/snowflake/spark/snowflake/FilterPushdown.scala

```diff
@@ -63,24 +63,24 @@ private[snowflake] object FilterPushdown {
       dataType match {
         case StringType =>
           StringVariable(
-            value.toString
+            Option(value).map(_.toString
               .replace("'", "''")
               .replace("\\", "\\\\")
-          ) !
+            )) !
         case DateType =>
-          StringVariable(value.asInstanceOf[Date].toString) + "::DATE"
+          StringVariable(Option(value).map(_.asInstanceOf[Date].toString)) + "::DATE"
         case TimestampType =>
-          StringVariable(value.asInstanceOf[Timestamp].toString) + "::TIMESTAMP(3)"
+          StringVariable(Option(value).map(_.asInstanceOf[Timestamp].toString)) + "::TIMESTAMP(3)"
```

Comment:
It looks it will cause some precision loss when timestamp is converted to "::TIMESTAMP(3)". I know it is not related to this fix. But I think it needs to be investigated.

Reply:
Yeah, this does seem suspect. Good catch. Let's investigate this, but outside of this PR.

Reply:
agree.